function/dataaccess/rawls.js

```
@@ -0,0 +1,54 @@
+'use strict';
+
+const {throwHttpError} = require('../errors');
+const restclient = require('./restclient');
+
+const bucketFromWorkspace = (ws) => {
+  return ws.bucketName;
+};
+
+const getBucket = async (appConfig, namespace, name, authToken) => {
+  const ws = await getWorkspace(appConfig, namespace, name, authToken);
+  return bucketFromWorkspace(ws);
+};
+
+const getWorkspace = async (appConfig, namespace, name, authToken) => {
+  const opts = appConfig || {};
+
+  if (!opts.rawlsUrl) {
+    throwHttpError(500, 'Rawls URL undefined; cannot continue.');
+  }
+
+  if (!namespace) {
+    throwHttpError(400, 'Workspace namespace is required.');
+  }
+
+  if (!name) {
+    throwHttpError(400, 'Workspace name is required.');
+  }
+
+  if (!authToken) {
+    throwHttpError(400, 'Authorization token is required.');
+  }
+
+  // query Rawls for this workspace
+  const reqOptions = {
+    uri: `${opts.rawlsUrl}/api/workspaces/${namespace}/${name}`,
+    successCodes: [200],
+    timeout: opts.timeout,
+  };
+
+  return await restclient.safeRequest(reqOptions, authToken)
+      .then((response) => {
+        // do we need "or PROJECT_OWNER" here?
+        if (response.body.accessLevel === 'OWNER' ||
+            response.body.accessLevel === 'PROJECT_OWNER') {
+          const ws = response.body.workspace || {};
```

If the response came back with a `200` and `response.body.workspace` is `falsey`, do you really want to just return an empty object like this?  Seems to me that if a call rawls to get a workspace responded with a 200 and was  missing the workspace, I'd rather throw an exception because that's weird.

---

good call on this -  I modified the code  to check for something vaguely valid from rawls.

---

so, back to you @gpolumbo-broad for re-review!